src/index.ts

```diff
@@ -81,22 +86,139 @@ export class Client {
       headers
     };
 
-    return fetchImpl(url, request).then(response => {
-      return response.json().then(
-        (body: SubmissionBody): SubmissionResponse => {
-          return { body, response };
+    // first check if we need to add the stripe paymentMethod
+    if (this.stripePromise && opts.createPaymentMethod) {
+      // Get Stripe payload
+      const payload = await opts.createPaymentMethod();
+
+      if (payload.error) {
+        // Return the error in case Stripe failed to create a payment method
+        return {
+          response: null,
+          body: {
+            errors: [
+              {
+                code: 'PAYMENT_METHOD_ERROR',
+                message: 'Error creating payment method',
+                field: 'paymentMethod'
+              }
+            ]
+          }
+        };
+      }
+
+      // Add the paymentMethod to the data
+      appendExtraData(data, 'paymentMethod', payload.paymentMethod.id);
+
+      // Send a request to Formspree server to handle the payment method
+      const response = await fetchImpl(url, {
+        ...request,
+        body: data
+      });
+      const responseData = await response.json();
+
+      // Handle server side errors
+      if (responseData.error) {
+        return {
+          response,
+          body: {
+            errors: responseData.errors
+          }
+        };
+      }
+
+      // Handle success response
+      if (responseData.next && responseData.ok) {
+        return {
+          response,
+          body: responseData
+        };
+      }
```

Comment:
Why do you need to check and catch these cases? In both the default success and error cases, you can just return

```
return { response, body: responseData };
```